mailpoet/lib/Cron/Workers/Scheduler.php

```diff
@@ -332,49 +340,56 @@ private function processReEngagementEmail($queue) {
     return true;
   }
 
-  public function verifyMailpoetSubscriber($subscriberId, $newsletter, $queue) {
-    $subscriber = Subscriber::findOne($subscriberId);
+  public function verifyMailpoetSubscriber($subscriberId, NewsletterEntity $newsletter, $queue) {
+    if (is_int($subscriberId)) {
+      $subscriber = $this->subscribersRepository->findOneById($subscriberId);
+    } else {
+      $subscriber = null;
+    }
+
     // check if subscriber is in proper segment
-    $subscriberInSegment =
-      SubscriberSegment::where('subscriber_id', $subscriberId)
-        ->where('segment_id', $newsletter->segment)
-        ->where('status', 'subscribed')
-        ->findOne();
+    $subscriberInSegment = $this->subscriberSegmentRepository->findOneBy(
+      [
+        'subscriber' => $subscriberId,
+        'segment' => $newsletter->getOptionValue('segment'),
+        'status' => SubscriberEntity::STATUS_SUBSCRIBED,
+      ]
+    );
     if (!$subscriber || !$subscriberInSegment) {
       $queue->delete();
       return false;
     }
     return $this->verifySubscriber($subscriber, $queue);
   }
 
-  public function verifyWPSubscriber($subscriberId, $newsletter, $queue) {
+  public function verifyWPSubscriber($subscriberId, NewsletterEntity $newsletter, $queue) {
     // check if user has the proper role
-    $subscriber = Subscriber::findOne($subscriberId);
-    if (!$subscriber || $subscriber->isWPUser() === false) {
+    $subscriber = $this->subscribersRepository->findOneById($subscriberId);
+    if (!$subscriber || $subscriber->isWPUser() === false || is_null($subscriber->getWpUserId())) {
       $queue->delete();
       return false;
     }
-    $wpUser = get_userdata($subscriber->wpUserId);
+    $wpUser = get_userdata($subscriber->getWpUserId());
     if ($wpUser === false) {
       $queue->delete();
       return false;
     }
     if (
-      $newsletter->role !== WelcomeScheduler::WORDPRESS_ALL_ROLES
-      && !in_array($newsletter->role, ((array)$wpUser)['roles'])
+      $newsletter->getOptionValue('role') !== WelcomeScheduler::WORDPRESS_ALL_ROLES
+      && !in_array($newsletter->getOptionValue('role'), ((array)$wpUser)['roles'])
     ) {
       $queue->delete();
       return false;
     }
     return $this->verifySubscriber($subscriber, $queue);
   }
 
-  public function verifySubscriber($subscriber, $queue) {
+  public function verifySubscriber(SubscriberEntity $subscriber, $queue) {
```

Comment:
```suggestion
  public function verifySubscriber(SubscriberEntity $subscriber, $queue): bool {
```

Reply:
Odd. Shouldn't code sniffer or some tool pick up missing return types?

Reply:
Maybe, I don't remember off the top of my head if we have a rule for that or not.